app/src/main/java/com/example/roomies/SettingsFragment.java

```diff
@@ -180,8 +195,40 @@ public void done(List<UserCircle> userCircles, ParseException e) {
                 }
 
                 // save received posts to list and notify adapter of new data
+                userCircle = userCircles.get(0);
```

Comment:
Can we add a sanity check here, to see if userCircles() is not empty? Assessing index without checking the size can easily lead to a indexOutOfBoundError.

Reply:
In the if statement before, the function checks if userCircles is empty. If it is, a new activity is started and this line is not reached.